care/utils/assetintegration/onvif.py

```diff
@@ -40,12 +182,58 @@ def handle_action(self, action):
             **action_data,
         }
 
+        if action_type == self.OnvifActions.LOCK_CAMERA.value:
+            if camera_manager.lock_camera():
+                return {
+                    "message": "You now have access to the camera controls, the camera is locked for other users",
+                    "camera_user": camera_manager.current_user(),
+                }
+
+            raise PermissionDenied(
+                {
+                    "message": "Camera is currently in used by another user, you have been added to the waiting list for camera controls access",
+                    "camera_user": camera_manager.current_user(),
+                }
+            )
+
+        if action_type == self.OnvifActions.UNLOCK_CAMERA.value:
+            camera_manager.unlock_camera()
+            return {"message": "Camera controls unlocked"}
+
+        if action_type == self.OnvifActions.REQUEST_ACCESS.value:
+            if camera_manager.request_access():
+                return {
+                    "message": "Access to camera camera controls granted",
+                    "camera_user": camera_manager.current_user(),
+                }
+
+            return {
+                "message": "Requested access to camera controls, waiting for current user to release",
+                "camera_user": camera_manager.current_user(),
+            }
+
+        if action_type == self.OnvifActions.GET_STREAM_TOKEN.value:
+            return self.api_post(
+                self.get_url("api/stream/getToken/videoFeed"),
+                {
+                    "stream_id": self.access_key,
+                },
+            )
+
         if action_type == self.OnvifActions.GET_CAMERA_STATUS.value:
             return self.api_get(self.get_url("status"), request_body)
 
         if action_type == self.OnvifActions.GET_PRESETS.value:
             return self.api_get(self.get_url("presets"), request_body)
 
+        if not camera_manager.has_access():
+            raise PermissionDenied(
+                {
+                    "message": "Camera is currently in used by another user, you have been added to the waiting list for camera controls access",
+                    "camera_user": camera_manager.current_user(),
+                }
+            )
```

Comment:
we can check this first and remove the other raise PermissionDenied( :192

Reply:
This check is only meant to block access to camera controls (like the ones involving moving the camera)